src/Database/Type/DecimalType.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.4
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+use Cake\Database\Driver;
+use Cake\Database\Type;
+use Cake\Database\TypeInterface;
+use PDO;
+use RuntimeException;
+
+/**
+ * Decimal type converter.
+ *
+ * Use to convert decimal data between PHP and the database types.
+ */
+class DecimalType extends Type implements TypeInterface
+{
+
+    /**
+     * Identifier name for this type
+     *
+     * @var string|null
+     */
+    protected $_name = null;
+
+    /**
+     * Constructor
+     *
+     * @param string|null $name The name identifying this type
+     */
+    public function __construct($name = null)
+    {
+        $this->_name = $name;
+    }
+
+    /**
+     * The class to use for representing number objects
+     *
+     * @var string
+     */
+    public static $numberClass = 'Cake\I18n\Number';
+
+    /**
+     * Whether numbers should be parsed using a locale aware parser
+     * when marshalling string inputs.
+     *
+     * @var bool
+     */
+    protected $_useLocaleParser = false;
+
+    /**
+     * Convert integer data into the database format.
+     *
+     * @param string|resource $value The value to convert.
+     * @param \Cake\Database\Driver $driver The driver instance to convert with.
+     * @return string|null
+     */
+    public function toDatabase($value, Driver $driver)
+    {
+        if ($value === null || $value === '') {
+            return null;
+        }
+        if (is_array($value)) {
+            return '1';
+        }
+        if (is_string($value) && is_numeric($value)) {
+            return $value;
+        }
+
+        return sprintf('%F', (float)$value);
+    }
+
+    /**
+     * Convert float values to PHP integers
```

Comment:
So because 'decimal' => 'Cake\Database\Type\DecimalType', pointed to float before this cannot/should not be fixed? Maybe we want something like BigNumType and have that behave with absolute precision and then map decimal there OPTIONALLY? Maybe it is a thing to be done for 4.0? However the docs should really have a fat warning that decimal type is not precise... I'd always expect decimal to behave like integer (or string) and never like float.